discord_thread_summarizer_using_openai/main.bal

```diff
@@ -0,0 +1,79 @@
+import ballerina/io;
+import ballerina/time;
+
+const string DISCORD_API_URL = "https://discord.com/api/v9";
+const TYPE_FORUM_CHANEL = 15;
+
+configurable string botToken = ?;
+configurable int forumChannelId = ?;
+
+function readThreads(ActiveThreads activeThreads) returns error? {
+
+    foreach ChannelThread thread in activeThreads.threads {
+
+        // check if the thread has any recent messages
+        if check checkRecentMessages(thread) {
+            string prompt = check constructPrompt(thread);
+            io:println(prompt);
+
+            string|error generateChatCompletionResult = generateChatCompletion(prompt, model = "gpt-3.5-turbo");
+
+            if generateChatCompletionResult is string {
+                io:println(generateChatCompletionResult);
+            } else {
+                io:println("Error: " + generateChatCompletionResult.toBalString());
```

Comment:
Rather than the `io` module, I feel that it is better to use `log` module to print errors with stack-trace.

Reply:
Addressed in https://github.com/ballerina-guides/ai-samples/pull/69/commits/8a038483b0bb440e4d9ac06074e5f2217a1309e3